common/src/main/java/tachyon/security/authentication/AuthenticationFactory.java

```diff
@@ -111,5 +118,27 @@ public static AuthType getAuthTypeFromConf(TachyonConf conf) {
         AuthType.NOSASL.getAuthName()));
   }
 
+  /**
+   * Create a new Thrift socket what will connect to the given address
+   * @param address The given address to connect
+   * @return An unconnected socket
+   */
+  public static TSocket createTSocket(InetSocketAddress address) {
+    return new TSocket(NetworkAddressUtils.getFqdnHost(address), address.getPort());
+  }
+
+  /**
+   * Create a ClientTransport for Plain
+   * @param username User Name of PlainClient
+   * @param password Password of PlainClient
+   * @param tTransport The original Transport
+   * @return Wrapped Transport with Plain
+   * @throws SaslException
+   */
+  public static TTransport createPlainClientTransport(String username, String password,
+      TTransport tTransport) throws SaslException {
+    return PlainSaslHelper.getPlainClientTransport(username, password, tTransport);
```

Comment:
Any reason we want this indirection? Can we either, implement `PlainSaslHelper#getPlainClientTransport` in `AuthenticationFactory` or let `PlainSaslHelper#getPlainClientTransport` to be called directly?